cupsfilters/texttotext.c

```diff
@@ -0,0 +1,1139 @@
+	    *(destptr ++) = *p;
+	  free(wrapped_word);
+	  wrapped_word = NULL;
+	  page_empty = 0;
+	  skip_spaces = 0;
+	}
+      }
+      if (*procptr == '\r' || *procptr == '\n') { /* CR or LF */
+	/* Only write newline if we are not on the LF of a CR+LF */
+	if (*procptr == '\r' || previous_is_cr == 0) {
+	  /* Remove trailing whitespace */
+	  while (destptr > out_page && *(destptr - 1) == ' ')
+	    destptr --;
+	  /* Put newline character(s) */
+	  for (j = 0; newline_char_str[j]; j ++)
+	    *(destptr ++) = newline_char_str[j];
+	  /* Position cursor in next line */
+	  line ++;
+	  column = 0;
+	  /* Finished truncating an overlong line */
+	  skip_rest_of_line = 0;
+	  skip_spaces = 0;
+	}
+	if (*procptr == '\r')
+	  previous_is_cr = 1;
+	else
+	  previous_is_cr = 0;
+      } else {
+	previous_is_cr = 0;
+	if (*procptr == '\t') { /* Tab character */
+	  if (!skip_rest_of_line && !skip_spaces) {
+	    *(destptr ++) = ' '; /* Always at least one space */
+	    column ++;
+	    /* Add spaces to reach next multiple of the tab width */
+	    for (; column % tab_width != 0 && column < text_width; column ++)
+	      *(destptr ++) = ' ';
+	  }
+	} else if (*procptr == '\f') { /* Form feed */
+	  /* Skip to end of page */	  
+	  if (send_ff)
+	    /* Mark page full */
+	    line = text_height;
+	  else if (pagination)
+	    /* Fill page with newlines */
+	    for (; line < text_height; line ++)
+	      for (j = 0; newline_char_str[j]; j ++)
+		*(destptr ++) = newline_char_str[j];
+	  column = 0;
+	  /* Finished truncating an overlong line */
+	  skip_rest_of_line = 0;
+	  skip_spaces = 0;
+	} else if (*procptr == ' ') { /* Space */
+	  if (!skip_rest_of_line && !skip_spaces) {
+	    *(destptr ++) = *procptr;
+	    column ++;
+	  }
+	} else if (*procptr > ' ' || *procptr < '\0') { /* Regular character */
+	  if (!skip_rest_of_line) {
+	    *(destptr ++) = *procptr;
+	    column ++;
+	    page_empty = 0;
+	    skip_spaces = 0;
+	  }
+	}
+      }
+    }
+  } while (outbuf[0] != '\0'); /* End of input file */
+
+  close(fd);
+  
+  if (iconv_close (cd) != 0)
+    if(log) log(ld, FILTER_LOGLEVEL_DEBUG, "texttotext: Error closing iconv encoding conversion session");
+
+  /* Error out on an illegal UTF-8 sequence in the input file */
+  if (result < 0)
+    goto error;
+
+  /* Print out the page array if we created one */
+  if (pagination &&
+      ((num_copies != 1 && collate) || reverse_order)) {
+    /* If we print collated copies, the outer loop (i) goes through the
+       copies, if we do not collate, the inner loop (j) goes through the
+       copies. The other loop only runs for one cycle. */
+    for (i = 0; i < (collate ? num_copies : 1); i ++)
+      for (page = (reverse_order ? num_pages : 1);
+	   (reverse_order ? (page >= 1) : (page <= num_pages));
+	   page += (reverse_order ? -1 : 1)) {
+	p = (char *)cupsArrayIndex(page_array, page - 1);
+	for (j = 0; j < (collate ? 1 : num_copies); j ++)
+	  printf("%s", p);
```

Comment:
What are these 2 lines good for? There should be nothing printed to stdout at all. The output should either got to outputfp if it belongs to the job or to the log function if it is logging.